app/js/registration_controller.js

```diff
@@ -3,6 +3,34 @@
 /* global User, Utils */
 
 (function(exports) {
+  var success_msg;
+  var confirmation_poll_timeout;
+
+  function check_for_email_confirmation() {
+    var promise = User.get_user(true);
+
+    return promise.then(function(user) {
+      if (user.is_active) {
+        success_msg.classList.add('hide');
+      }
+      return user.is_active;
+    });
+  }
+
+  function poll_for_email_confirmation() {
+    confirmation_poll_timeout = window.setTimeout(function() {
+      var promise = check_for_email_confirmation();
+
+      promise.then(function(active) {
+        if (active) {
```

Comment:
No need for this branch since `check_for_email_confirmation` already hides the overlay.